scheduler/recovery.go

```diff
@@ -40,13 +83,51 @@ func (rd *RecoveryDescriptor) Run(ctx context.Context, pc chan<- Process) error
 		} else if !cont {
 			break
 		}
+
```

Comment:
I think you can just handle the retry counter in this method:


```go

func (rd *RecoveryDescriptor) Run(ctx context.Context, pc chan<- Process) error {
  var currentRetries int
  for {
    start := time.Now()
    // Call runOnce, etc.
    if time.Now().Sub(start) >= rd.resetDuration {
      currentRetries = 0
    }
    // Check as before...
  }
}
```

But I may be missing something obvious here, fair warning as usual. :-)